cpp/models/ode_seir/model.h

```diff
@@ -48,51 +48,49 @@ using Flows = TypeList<Flow<InfectionState::Susceptible, InfectionState::Exposed
-
-            for (auto j = AgeGroup(0); j < n_agegroups; j++) {
-
-                size_t Sj = this->populations.get_flat_index({j, InfectionState::Susceptible});
-                size_t Ej = this->populations.get_flat_index({j, InfectionState::Exposed});
-                size_t Ij = this->populations.get_flat_index({j, InfectionState::Infected});
-                size_t Rj = this->populations.get_flat_index({j, InfectionState::Recovered});
-
-                double Nj    = pop[Sj] + pop[Ej] + pop[Ij] + pop[Rj];
-                double divNj = 1.0 / Nj;
-
-                double coeffStoE = contact_matrix.get_matrix_at(t)(static_cast<Eigen::Index>((size_t)i),
-                                                                   static_cast<Eigen::Index>((size_t)j)) *
-                                   params.get<TransmissionProbabilityOnContact>()[i] * divNj;
-
-                double dummy_S = y[Si] * y[Ij] * coeffStoE;
-
-                flows[get_flat_flow_index<InfectionState::Susceptible, InfectionState::Exposed>({i})] += dummy_S;
+        const Index<AgeGroup> age_groups = reduce_index<Index<AgeGroup>>(this->populations.size());
+        const auto& params               = this->parameters;
+
+        for (auto j : make_index_range(age_groups)) {
+            const size_t Sj = this->populations.get_flat_index({j, InfectionState::Susceptible});
+            const size_t Ej = this->populations.get_flat_index({j, InfectionState::Exposed});
+            const size_t Ij = this->populations.get_flat_index({j, InfectionState::Infected});
+            const size_t Rj = this->populations.get_flat_index({j, InfectionState::Recovered});
+
+            const double Nj_inv = 1.0 / (pop[Sj] + pop[Ej] + pop[Ij] + pop[Rj]);
+
+            for (auto i : make_index_range(age_groups)) {
+                const size_t Si = this->populations.get_flat_index({i, InfectionState::Susceptible});
+                const size_t Ei = this->populations.get_flat_index({i, InfectionState::Exposed});
+                const size_t Ii = this->populations.get_flat_index({i, InfectionState::Infected});
+
+                const double coeffStoE =
+                    params.template get<ContactPatterns<FP>>().get_cont_freq_mat().get_matrix_at(t)(i.get(), j.get()) *
+                    params.template get<TransmissionProbabilityOnContact<FP>>()[i] * Nj_inv;
+
+                flows[Base::template get_flat_flow_index<InfectionState::Susceptible, InfectionState::Exposed>(i)] +=
+                    coeffStoE * y[Si] * y[Ij];
+                flows[Base::template get_flat_flow_index<InfectionState::Exposed, InfectionState::Infected>(i)] =
+                    (1.0 / params.template get<TimeExposed<FP>>()[i]) * y[Ei];
+                flows[Base::template get_flat_flow_index<InfectionState::Infected, InfectionState::Recovered>(i)] =
+                    (1.0 / params.template get<TimeInfected<FP>>()[i]) * y[Ii];
```

Comment:
These two flows are independent of the contact age group i. Currently, we calculate these flows far too often.
```suggestion
                }
            flows[Base::template get_flat_flow_index<InfectionState::Exposed, InfectionState::Infected>(j)] =
                (1.0 / params.template get<TimeExposed<FP>>()[j]) * y[Ej];
            flows[Base::template get_flat_flow_index<InfectionState::Infected, InfectionState::Recovered>(i)] =
                (1.0 / params.template get<TimeInfected<FP>>()[j]) * y[Ij];
```

Reply:
You are absolutely right. I will fix that.